server/src/main/java/org/apache/iotdb/db/metadata/mnode/InternalMNode.java

```diff
@@ -46,13 +51,34 @@ public void addChild(String name, MNode child) {
 
 
   @Override
-  public void deleteChild(String name) {
-    children.remove(name);
-  }
+  public void deleteChild(String name) throws DeleteFailedException {
```

Comment:
add java doc.

If delete a leafNode, lock its parent, if delete an InternalNode, lock this node